### ISSUE TITLE
Music volume issue

### ISSUE THREAD

**Issue by ETHGRE** [RHeavenStudio/HeavenStudio#494]
### Describe the bug

Whenever you set down a music volume marker, even after removing it, the volume will still change when it passes the marker's position.

### How to reproduce the bug?

Just place down a music volume marker and remove it. If that doesn't cause the bug, play the music past the marker first, remove it, and then play the music up to that point again. The volume change still applies.

### Screenshots/Videos

https://github.com/RHeavenStudio/HeavenStudio/assets/137836357/0f192d11-ae29-43b4-b23c-3a288eef5bd9

### What Operating System(s) were you using?

Windows 10/11

### Log file

```shell
Mono path[0] = 'C:/Users/ethan/Downloads/StandaloneWindows64-build/Heaven Studio_Data/Managed'
Mono config path = 'C:/Users/ethan/Downloads/StandaloneWindows64-build/MonoBleedingEdge/etc'
Initialize engine version: 2021.3.21f1 (1b156197d683)
[Subsystems] Discovering subsystems at path C:/Users/ethan/Downloads/StandaloneWindows64-build/Heaven Studio_Data/UnitySubsystems
GfxDevice: creating device client; threaded=1; jobified=1
Direct3D:
    Version:  Direct3D 11.0 [level 11.1]
    Renderer: Intel(R) UHD Graphics 600 (ID=0x3185)
    Vendor:   Intel
    VRAM:     1960 MB
    Driver:   30.0.101.1340
Begin MonoManager ReloadAssembly
- Completed reload, in  0.327 seconds
<RI> Initializing input.
<RI> Input initialized.
<RI> Initialized touch support.
UnloadTime: 3.130800 ms
Flushing possible JoyShocks...
No JoyShocks found.
The character used for Underline is not available in font asset [rodin_lat_cy_ja_ko_spec SDF].
Unloading 6 Unused Serialized files (Serialized files now loaded: 0)
UnloadTime: 10.863700 ms
Running game loader RvlBadmintonLoader
Running game loader CtrBearLoader
Running game loader RvlBoardMeetingLoader
Running game loader NtrFlickLoader
Running game loader CtrCatchLoader
Running game loader RvlBookLoader
Running game loader AgbClapLoader
Running game loader NtrCoinLoader
Running game loader NtrCropLoader
Running game loader NtrDjLoader
Running game loader NtrDogNinjaLoader
Running game loader RvlDoubleDateLoader
Running game loader CtrDrummingLoader
Running game loader NtrIdolLoader
Running game loader AgbFireworkLoader
Running game loader CtrFirstContact
Running game loader RvlFlipperFlopLoader
Running game loader RvlForkLoader
Running game loader NtrGleeClubLoader
Running game loader RvlNewKarateLoader
Running game loader CtrTeppanLoader
Running game loader RvlRocketLoader
Running game loader NtrBackbeatLoader
Running game loader AgbMarcherLoader
Running game loader pcoMeatLoader
Running game loader AgbUpbeatLoader
Running game loader ntrMunchyMonkLoader
Running game loader NtrOctopusMachineLoader
Running game loader CtrPillowLoader
Running game loader AgbQuizShowLoader
Running game loader NtrPingpongLoader
Running game loader PcoSomenLoader
Running game loader AgbHairLoader
Running game loader RvlRingsideLoader
Running game loader NtrRockersLoader
Running game loader NtrSamuraiLoader
Running game loader RvlSeeSawLoader
Running game loader AgbGhostLoader
Running game loader AgbSpaceDanceLoader
Running game loader NtrSoccerLoader
Running game loader AgbBatterLoader
Running game loader RvlTambourineLoader
Running game loader AgbTapLoader
Running game loader RvlTapTroupeLoader
Running game loader NtrDazzlesLoader
Running game loader AgbTossBoysLoader
Running game loader AgbTramLoader
Running game loader MobTrickLoader
Running game loader NtrTunnelLoader
Running game loader AgbWaltzLoader
Running game loader RvlWorkingDoughLoader
ResultException: InternalError
  at Discord.Discord..ctor (System.Int64 clientId, System.UInt64 flags) [0x00327] in <c19283e3adb8481d9e29addb7f7d75ee>:0 
  at HeavenStudio.DiscordRPC.DiscordController.Connect () [0x00000] in <22cfb65588394938ab8e490c0cc76b8a>:0 
  at HeavenStudio.DiscordRPC.DiscordRPC.DiscordControllerCheck () [0x00022] in <22cfb65588394938ab8e490c0cc76b8a>:0 
  at HeavenStudio.DiscordRPC.DiscordRPC.UpdateActivity (System.String state, System.String details, System.Boolean updateTime) [0x00000] in <22cfb65588394938ab8e490c0cc76b8a>:0 
  at HeavenStudio.GlobalGameManager.UpdateDiscordStatus (System.String details, System.Boolean editor, System.Boolean updateTime) [0x0001a] in <22cfb65588394938ab8e490c0cc76b8a>:0 
  at HeavenStudio.Editor.Editor.UpdateEditorStatus (System.Boolean updateTime) [0x0000a] in <22cfb65588394938ab8e490c0cc76b8a>:0 
  at HeavenStudio.Editor.Editor.Init () [0x001c3] in <22cfb65588394938ab8e490c0cc76b8a>:0 
  at HeavenStudio.GameInitializer.Start () [0x00211] in <22cfb65588394938ab8e490c0cc76b8a>:0 

Create TimingDisplayComponent
Create SkillStarComponent
Create SectionComponent
Unloading 595 unused Assets to reduce memory usage. Loaded Objects now: 6983.
Total: 41.396600 ms (FindLiveObjects: 4.134300 ms CreateObjectMapping: 2.005100 ms MarkObjects: 33.959700 ms  DeleteObjects: 1.296300 ms)

loaded song
Playing at 0
== Playthrough statistics of New Remix (played at 2023-06-26 22:47:15) ==
Average input offset for playthrough: 0ms
Accuracy for playthrough:  0.00
Cleared 0 sections out of 0
Skill Star not collected
Perfect Clear!
Playing at 0
== Playthrough statistics of New Remix (played at 2023-06-26 22:47:23) ==
Average input offset for playthrough: 0ms
Accuracy for playthrough:  0.00
Cleared 0 sections out of 0
Skill Star not collected
Perfect Clear!
Disconnecting JoyShocks...
Clearing RIQ Cache...
Setting up 1 worker threads for Enlighten.
Memory Statistics:
[ALLOC_TEMP_TLS] TLS Allocator
  StackAllocators : 
    [ALLOC_TEMP_MAIN]
      Peak usage frame count: [0-1.0 KB]: 785 frames, [2.0 KB-4.0 KB]: 1 frames, [8.0 KB-16.0 KB]: 59 frames, [16.0 KB-32.0 KB]: 140 frames, [32.0 KB-64.0 KB]: 731 frames, [64.0 KB-128.0 KB]: 1 frames, [256.0 KB-0.5 MB]: 1 frames, [0.5 MB-1.0 MB]: 1 frames, [2.0 MB-4.0 MB]: 1 frames
      Initial Block Size 4.0 MB
      Current Block Size 4.0 MB
      Peak Allocated Bytes 2.9 MB
      Overflow Count 0
    [ALLOC_TEMP_Loading.AsyncRead]
      Initial Block Size 64.0 KB
      Current Block Size 64.0 KB
      Peak Allocated Bytes 288 B
      Overflow Count 0
    [ALLOC_TEMP_Loading.PreloadManager]
      Initial Block Size 256.0 KB
      Current Block Size 492.0 KB
      Peak Allocated Bytes 444.7 KB
      Overflow Count 17
    [ALLOC_TEMP_Background Job.Worker 8]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 54 B
      Overflow Count 0
    [ALLOC_TEMP_Background Job.Worker 9]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 54 B
      Overflow Count 0
    [ALLOC_TEMP_Job.Worker 0]
      Initial Block Size 256.0 KB
      Current Block Size 256.0 KB
      Peak Allocated Bytes 37.8 KB
      Overflow Count 0
    [ALLOC_TEMP_Background Job.Worker 10]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 54 B
      Overflow Count 0
    [ALLOC_TEMP_Background Job.Worker 14]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 54 B
      Overflow Count 0
    [ALLOC_TEMP_Background Job.Worker 6]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 54 B
      Overflow Count 0
    [ALLOC_TEMP_Background Job.Worker 12]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 54 B
      Overflow Count 0
    [ALLOC_TEMP_EnlightenWorker]
      Initial Block Size 64.0 KB
      Current Block Size 64.0 KB
      Peak Allocated Bytes 54 B
      Overflow Count 0
    [ALLOC_TEMP_Background Job.Worker 15]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 54 B
      Overflow Count 0
    [ALLOC_TEMP_Background Job.Worker 1]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 54 B
      Overflow Count 0
    [ALLOC_TEMP_Background Job.Worker 2]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 54 B
      Overflow Count 0
    [ALLOC_TEMP_Background Job.Worker 7]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 54 B
      Overflow Count 0
    [ALLOC_TEMP_UnityGfxDeviceWorker]
      Initial Block Size 256.0 KB
      Current Block Size 256.0 KB
      Peak Allocated Bytes 6.8 KB
      Overflow Count 0
    [ALLOC_TEMP_AssetGarbageCollectorHelper]
      Initial Block Size 64.0 KB
      Current Block Size 64.0 KB
      Peak Allocated Bytes 0 B
      Overflow Count 0
    [ALLOC_TEMP_Background Job.Worker 5]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 54 B
      Overflow Count 0
    [ALLOC_TEMP_Background Job.Worker 13]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 54 B
      Overflow Count 0
    [ALLOC_TEMP_Background Job.Worker 11]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 54 B
      Overflow Count 0
    [ALLOC_TEMP_Background Job.Worker 3]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 54 B
      Overflow Count 0
    [ALLOC_TEMP_Background Job.Worker 0]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 240 B
      Overflow Count 0
    [ALLOC_TEMP_Background Job.Worker 4]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 54 B
      Overflow Count 0
    [ALLOC_TEMP_BatchDeleteObjects]
      Initial Block Size 64.0 KB
      Current Block Size 64.0 KB
      Peak Allocated Bytes 54 B
      Overflow Count 0
[ALLOC_DEFAULT] Dual Thread Allocator
  Peak main deferred allocation count 930
    [ALLOC_BUCKET]
      Large Block size 4.0 MB
      Used Block count 1
      Peak Allocated bytes 3.2 MB
    [ALLOC_DEFAULT_MAIN]
      Peak usage frame count: [8.0 MB-16.0 MB]: 847 frames, [16.0 MB-32.0 MB]: 873 frames
      Requested Block Size 16.0 MB
      Peak Block count 2
      Peak Allocated memory 18.2 MB
      Peak Large allocation bytes 0 B
    [ALLOC_DEFAULT_THREAD]
      Peak usage frame count: [16.0 MB-32.0 MB]: 911 frames, [32.0 MB-64.0 MB]: 809 frames
      Requested Block Size 16.0 MB
      Peak Block count 1
      Peak Allocated memory 50.2 MB
      Peak Large allocation bytes 46.6 MB
[ALLOC_TEMP_JOB_1_FRAME]
  Initial Block Size 2.0 MB
  Used Block Count 1
  Overflow Count (too large) 0
  Overflow Count (full) 0
[ALLOC_TEMP_JOB_2_FRAMES]
  Initial Block Size 2.0 MB
  Used Block Count 1
  Overflow Count (too large) 0
  Overflow Count (full) 0
[ALLOC_TEMP_JOB_4_FRAMES (JobTemp)]
  Initial Block Size 2.0 MB
  Used Block Count 3
  Overflow Count (too large) 0
  Overflow Count (full) 0
[ALLOC_TEMP_JOB_ASYNC (Background)]
  Initial Block Size 1.0 MB
  Used Block Count 3
  Overflow Count (too large) 0
  Overflow Count (full) 0
[ALLOC_GFX] Dual Thread Allocator
  Peak main deferred allocation count 1
    [ALLOC_BUCKET]
      Large Block size 4.0 MB
      Used Block count 1
      Peak Allocated bytes 3.2 MB
    [ALLOC_GFX_MAIN]
      Peak usage frame count: [1.0 MB-2.0 MB]: 846 frames, [2.0 MB-4.0 MB]: 1 frames, [4.0 MB-8.0 MB]: 873 frames
      Requested Block Size 16.0 MB
      Peak Block count 1
      Peak Allocated memory 5.3 MB
      Peak Large allocation bytes 0 B
    [ALLOC_GFX_THREAD]
      Peak usage frame count: [128.0 KB-256.0 KB]: 848 frames, [256.0 KB-0.5 MB]: 1 frames, [0.5 MB-1.0 MB]: 871 frames
      Requested Block Size 16.0 MB
      Peak Block count 1
      Peak Allocated memory 0.8 MB
      Peak Large allocation bytes 0 B
[ALLOC_CACHEOBJECTS] Dual Thread Allocator
  Peak main deferred allocation count 751
    [ALLOC_BUCKET]
      Large Block size 4.0 MB
      Used Block count 1
      Peak Allocated bytes 3.2 MB
    [ALLOC_CACHEOBJECTS_MAIN]
      Peak usage frame count: [8.0 MB-16.0 MB]: 1720 frames
      Requested Block Size 4.0 MB
      Peak Block count 2
      Peak Allocated memory 15.9 MB
      Peak Large allocation bytes 9.7 MB
    [ALLOC_CACHEOBJECTS_THREAD]
      Peak usage frame count: [1.0 MB-2.0 MB]: 1700 frames, [2.0 MB-4.0 MB]: 20 frames
      Requested Block Size 4.0 MB
      Peak Block count 2
      Peak Allocated memory 3.9 MB
      Peak Large allocation bytes 0 B
[ALLOC_TYPETREE] Dual Thread Allocator
  Peak main deferred allocation count 0
    [ALLOC_BUCKET]
      Large Block size 4.0 MB
      Used Block count 1
      Peak Allocated bytes 3.2 MB
    [ALLOC_TYPETREE_MAIN]
      Peak usage frame count: [16.0 KB-32.0 KB]: 847 frames, [32.0 KB-64.0 KB]: 873 frames
      Requested Block Size 2.0 MB
      Peak Block count 1
      Peak Allocated memory 35.2 KB
      Peak Large allocation bytes 0 B
    [ALLOC_TYPETREE_THREAD]
      Peak usage frame count: [4.0 KB-8.0 KB]: 845 frames, [8.0 KB-16.0 KB]: 875 frames
      Requested Block Size 2.0 MB
      Peak Block count 1
      Peak Allocated memory 9.3 KB
      Peak Large allocation bytes 0 B
```

**Comment by Firerise9402** [comment]
I've found a solution for this I think, although it's quite weird. For some reason, markers like song volume and sections will place down two instead of one, with one being invisible. What you have to do is delete one, save, reboot heaven studio, and then open the riq again. The markers will be visible then if this is anything like what I encountered, and can then be deleted for good.

**Comment by minenice55** [comment]
can't seem to repro, though I do notice a discord RPC crash in the log that may have halted some part of the editor

**Comment by minenice55** [comment]
found cause of issue, will be fixed in #454
